src/Chapter2.hs

```diff
@@ -857,7 +875,13 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+--rewind [] = []
+rewind x = go [] x
+    where
+        go :: [a] -> [a] -> [a]
+        go rArr [] = rArr
+        go rArr (x:xs) = go (x : rArr) xs
```

Comment:
Amazing and most importantly efficient solution on the advanced task! Very well thought 👍🏼